rust/src/composepost.rs

```diff
@@ -904,6 +910,56 @@ fn hardlink_rpmdb_base_location(
     Ok(true)
 }
 
+#[context("Rewriting rpmdb for target native format")]
+fn rewrite_rpmdb_for_target_inner(rootfs_dfd: &openat::Dir) -> Result<()> {
+    let tempetc = crate::core::prepare_tempetc_guard(rootfs_dfd.as_raw_fd())?;
+
+    let dbfd = Rc::new(
+        memfd::MemfdOptions::default()
+            .allow_sealing(true)
+            .create("rpmdb")?
+            .into_file(),
+    );
+
+    let dbpath_arg = format!("--dbpath=/proc/self/cwd/{}", RPMOSTREE_RPMDB_LOCATION);
+    // Fork rpmdb from the *host* rootfs to read the rpmdb back into memory
+    let r = Exec::cmd("rpmdb")
+        .args(&[dbpath_arg.as_str(), "--exportdb"])
+        .cwd(format!("/proc/self/fd/{}", rootfs_dfd.as_raw_fd()))
+        .stdout(Redirection::RcFile(Rc::clone(&dbfd)))
+        .join()?;
+    if !r.success() {
+        return Err(anyhow!("Failed to execute rpmdb --exportdb: {:?}", r));
+    }
+
+    // Clear out the db on disk
+    rootfs_dfd.remove_all(RPMOSTREE_RPMDB_LOCATION)?;
+    rootfs_dfd.create_dir(RPMOSTREE_RPMDB_LOCATION, 0o755)?;
+
+    // Only one owner now
+    let mut dbfd = Rc::try_unwrap(dbfd).unwrap();
+    dbfd.seek(std::io::SeekFrom::Start(0))?;
```

Comment:
(Incidentally, not going to lie, it took me a fair bit of time to realize the reason the target rpmdb was empty was because I was forgetting this `seek` call :cry: )